From 5b648f54eb6d3c5d74372a4125e46d79799018f0 Mon Sep 17 00:00:00 2001
From: Benoit Parrot <bparrot@ti.com>
Date: Thu, 16 Sep 2021 16:26:08 +0200
Subject: [PATCH] ARM: dts: dra7: Add ti-sysc node for VIP1

Add VIP1 node as a child of l4 interconnect in order for it to probe
using ti-sysc.

Signed-off-by: Benoit Parrot <bparrot@ti.com>
Signed-off-by: Bartosz Golaszewski <bgolaszewski@baylibre.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 arch/arm/boot/dts/dra7-l4.dtsi | 43 +++++++++++++++++++++++++++++++++-
 1 file changed, 42 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/dra7-l4.dtsi b/arch/arm/boot/dts/dra7-l4.dtsi
index c1405f84e5d2..e67be859f522 100644
--- a/arch/arm/boot/dts/dra7-l4.dtsi
+++ b/arch/arm/boot/dts/dra7-l4.dtsi
@@ -4110,7 +4110,48 @@ target-module@170000 {			/* 0x48970000, ap 21 0a.0 */
 			#address-cells = <1>;
 			#size-cells = <1>;
 			ranges = <0x0 0x170000 0x10000>;
-			status = "disabled";
+
+			vip1: vip@0 {
+				compatible = "ti,dra7-vip1";
+				reg = <0x0000 0x114>,
+				      <0x5500 0xd8>,
+				      <0x5700 0x18>,
+				      <0x5800 0x80>,
+				      <0x5a00 0xd8>,
+				      <0x5c00 0x18>,
+				      <0x5d00 0x80>,
+				      <0xd000 0x400>;
+				reg-names = "vip",
+					    "parser0",
+					    "csc0",
+					    "sc0",
+					    "parser1",
+					    "csc1",
+					    "sc1",
+					    "vpdma";
+				interrupts = <GIC_SPI 351 IRQ_TYPE_LEVEL_HIGH>,
+					     <GIC_SPI 392 IRQ_TYPE_LEVEL_HIGH>;
+				/* CTRL_CORE_SMA_SW_1 */
+				ti,vip-clk-polarity = <&scm_conf 0x534>;
+
+				ports {
+					#address-cells = <1>;
+					#size-cells = <0>;
+
+					vin1a: port@0 {
+						reg = <0>;
+					};
+					vin2a: port@1 {
+						reg = <1>;
+					};
+					vin1b: port@2 {
+						reg = <2>;
+					};
+					vin2b: port@3 {
+						reg = <3>;
+					};
+				};
+			};
 		};
 
 		target-module@190000 {			/* 0x48990000, ap 23 2e.0 */
-- 
2.32.0 (Apple Git-132)

